src/ai/chatAi.ts

```diff
@@ -46,6 +52,30 @@ export default class ChatAi {
     return this.ain.sendTransaction(txBody);
   }
 
+  async depositCredit(
+    provider: string,
+    api: string,
+    amount: number
+  ): Promise<CreditDepositTransactionResult> {
+    const address = this.ain.signer.getAddress();
+
+    const aiName = validateAndGetAiName(provider, api);
+    const aiService = await validateAndGetAiService(aiName, this.ainize);
+
+    await loginAiService();
```

Comment:
I'm not sure it is good way..

Reply:
What about logging in once somewhere and keeping the session til the end?

Reply:
I think ainize-js need to provide two versions of chargeCredit(), i.e., the current version is safe for backend use only so it needs to provide another version for frontend use, e.g., getChargeCreditTxBody() so that the frontend can sign and send it directly to the blockchain.

In that sense, the current structure needs to be upgraded once we have the frontend version of the ainize API.

I think it's up to you to merge it with the current version now and make progress or wait until they provide that API.

Reply:
let's discuss this in today's meeting. thanks!

Reply:
@platfowner @minsulee2 As discussed with Ainize team, I will update this login function to use signer if implemented in ainize-js.